zap/src/main/java/org/zaproxy/zap/authentication/AuthenticationMethod.java

```diff
@@ -193,31 +289,126 @@ public boolean isAuthenticated(HttpMessage msg) {
+                            break;
+                        case REQUESTS:
+                        default:
+                            if (requestsSincePoll < pollFrequency) {
+                                requestsSincePoll++;
+                                try {
+                                    Stats.incCounter(
+                                            SessionStructure.getHostName(msg),
+                                            AUTH_STATE_ASSUMED_IN_STATS);
+                                } catch (URIException e) {
+                                    // Ignore
+                                }
+                                return true;
+                            }
+                            break;
+                    }
+                }
+                // Make the poll request
+                try {
+                    HttpMessage pollMsg = new HttpMessage(new URI(this.getPollUrl(), true));
+                    if (this.getPollData() != null && this.getPollData().length() > 0) {
+                        pollMsg.getRequestHeader().setMethod(HttpRequestHeader.POST);
+                        pollMsg.getRequestBody().setBody(this.getPollData());
+                        pollMsg.getRequestHeader()
+                                .setContentLength(pollMsg.getRequestBody().length());
+                    }
+                    if (this.getPollHeaders() != null && this.getPollHeaders().length() > 0) {
+                        for (String header : this.getPollHeaders().split("\n")) {
+                            String[] headerValue = header.split(":");
+                            if (headerValue.length == 2) {
+                                pollMsg.getRequestHeader()
+                                        .addHeader(headerValue[0].trim(), headerValue[1].trim());
+                            } else {
+                                LOGGER.error(
+                                        "Invalid header '"
+                                                + header
+                                                + "' for poll request to "
+                                                + this.getPollUrl());
+                            }
+                        }
+                    }
+                    pollMsg.setRequestingUser(user);
+                    if (this.getType() != null && user != null) {
+                        this.getType().replaceUserDataInPollRequest(pollMsg, user);
+                    }
+
+                    getHttpSender().sendAndReceive(pollMsg);
+                    AuthenticationHelper.addAuthMessageToHistory(pollMsg);
+                    contentToTest.add(pollMsg.getResponseHeader().toString());
+                    contentToTest.add(pollMsg.getResponseBody().toString());
+                    lastPollTime = new Date();
+                    requestsSincePoll = 0;
+
+                } catch (Exception e1) {
+                    LOGGER.error(e1.getMessage(), e1);
```

Comment:
Warn, this is expected to happen in case of connection problems, also, worth adding a message saying that it failed to send the auth poll request.